after/syntax/sml.vim

```diff
@@ -13,60 +13,69 @@
 "
 "       au! FileType sml setlocal conceallevel=2
 "
-
 " Fix wonky highlighting for => operator
 syn match smlOperator  "=>"
 
 " Also highlight 'NOTE' and 'Note' in comments
 syn keyword  smlTodo contained NOTE Note
 
-" Highlight type variables (i.e., tokens that look like 'xyz)
-syn match smlType "\<'\w\+\>"
-
-" Add conceal characters for common greek letters
-" that are used for type variables
-syn match smlType "'a\>"               conceal cchar=α
-syn match smlType "'alpha\>"           conceal cchar=α
-syn match smlType "'b\>"               conceal cchar=β
-syn match smlType "'beta\>"            conceal cchar=β
-syn match smlType "'c\>"               conceal cchar=γ
-syn match smlType "'gamma\>"           conceal cchar=γ
-syn match smlType "'d\>"               conceal cchar=δ
-syn match smlType "'delta\>"           conceal cchar=δ
-syn match smlType "'e\>"               conceal cchar=ε
-syn match smlType "'epsilon\>"         conceal cchar=ε
-
-syn match smlType "'zeta\>"            conceal cchar=ζ
-syn match smlType "'eta\>"             conceal cchar=η
-syn match smlType "'theta\>"           conceal cchar=θ
-syn match smlType "'kappa\>"           conceal cchar=κ
-syn match smlType "'lambda\>"          conceal cchar=λ
-
-syn match smlType "'m\>"               conceal cchar=μ
-syn match smlType "'mu\>"              conceal cchar=μ
-syn match smlType "'n\>"               conceal cchar=ν
-syn match smlType "'nu\>"              conceal cchar=ν
-
-syn match smlType "'xi\>"              conceal cchar=ξ
-
-syn match smlType "'p\>"               conceal cchar=π
-syn match smlType "'pi\>"              conceal cchar=π
-syn match smlType "'r\>"               conceal cchar=ρ
-syn match smlType "'rho\>"             conceal cchar=ρ
-syn match smlType "'s\>"               conceal cchar=σ
-syn match smlType "'sigma\>"           conceal cchar=σ
-syn match smlType "'t\>"               conceal cchar=τ
-syn match smlType "'tau\>"             conceal cchar=τ
-
-syn match smlType "'upsilon\>"         conceal cchar=υ
-syn match smlType "'phi\>"             conceal cchar=ϕ
-syn match smlType "'x\>"               conceal cchar=χ
-syn match smlType "'chi\>"             conceal cchar=χ
-
-syn match smlType "'psi\>"             conceal cchar=ψ
-
-syn match smlType "'w\>"               conceal cchar=ω
-syn match smlType "'omega\>"           conceal cchar=ω
+" Add conceal characters for common Greek letters
+" that are used for type variables.
+function! s:ConcealTyvar(tyvar, cchar)
+    let l:tick = "'"
+    if exists('g:sml_greek_tyvar_show_tick') && g:sml_greek_tyvar_show_tick
+        let l:tick = '\%(' . l:tick . '\)\@<='
```

Comment:
The regex, as the comment says, ends up looking like `\%('\)\@<=a\>`. This is
```
\%(             start of a non-capturing group
   '            that contains a tick
    \)          and ends here,
      \@<=      used for a positive lookbehind assertion
          a     followed by U+0061
           \>   and then a rightward word boundary
```
The construct `\%(expr\)` is a non-capturing group, which you perhaps know as `(?:expr)` in other regex flavors. Suppose you have something like `(...) (...)-(....)`; you could use this to reformat a phone number like `(555) 123-4567` as `555.123.4567` with the replacement `\1.\2.\3`, using backreferences. This is because each group _captures_ its argument to a reference. Using the `(?:expr)` construct (vim's `\%(expr\)`) omits this, freeing up space and making the parsing a bit faster. You should use it whenever you just want to group something logically without referencing it later.

I technically don't need it here, because the tick is already a single atom. But it's probably good to include anyway, in case someone wants to add support for equality types (`''Z`) or do anything else that would make the backreference target non-atomic.